src/ErrorHandler.php

```diff
@@ -54,11 +63,33 @@ public function addFormatters(FormatterInterface ...$formatters): self
         return $this;
     }
 
+    /**
+     * Set a custom log callback
+     */
+    public function logCallback(callable $callback): self
+    {
+        $this->logCallback = $callback;
+
+        return $this;
+    }
+
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
         try {
             return $handler->handle($request);
         } catch (Throwable $error) {
+            if ($this->logger) {
+                if ($this->logCallback) {
+                    ($this->logCallback)($this->logger, $error, $request);
```

Comment:
If the logger is callback, I'd do `$this->logger($error, $request)`. The callback is a callable with access to a logger (or anything else).